packages/datatrak-web/src/components/FullScreenSelect.tsx

```diff
@@ -0,0 +1,231 @@
+    <Slide direction="left" ref={ref} {...props} />
+  ),
+);
+
+const Modal = styled(Dialog).attrs({
+  PaperComponent: Fragment, // Extra wrapper is semantically meaningless, interferes with layout
+  TransitionComponent: Transition,
+  disableElevation: true,
+  fullScreen: true,
+  mountOnEnter: true,
+  unmountOnExit: true,
+})`
+  .MuiDialog-container {
+    background-color: ${({ theme }) => theme.palette.background.default};
+    display: grid;
+    grid-template-rows: auto 1fr;
+  }
+  .MuiDialog-scrollPaper {
+    justify-content: initial;
+    align-items: initial;
+  }
+`;
+
+const StyledDialogTitle = styled(DialogTitle).attrs({ disableTypography: true })`
+  display: contents;
+  padding: 0;
+`;
+const StyledDialogContent = styled(DialogContent).attrs({ dividers: true })`
+  overflow-y: auto;
+  padding-block-end: env(safe-area-inset-bottom, 0);
+  padding-block-start: 0;
+  padding-inline: 0;
+`;
+
+const Header = styled(StickyMobileHeader)`
+  inset-block-start: initial;
+  inset-inline-start: initial;
+  position: initial;
+  z-index: initial;
+`;
+
+const StyledList = styled(List).attrs({
+  disablePadding: true,
+  role: 'radiogroup',
+})`
+  & + * {
+    margin-block-start: 1.5rem;
+  }
+`;
+const StyledListItem = styled(ListItem).attrs({
+  button: true,
+  divider: true,
+  role: 'radio',
+})`
+  background-color: ${({ theme }) => theme.palette.background.paper};
+  display: grid;
+  grid-template-columns: 1fr minmax(1.5rem, auto);
+  grid-template-rows: minmax(1.5rem, auto);
+  padding-block: 0.75rem;
+  padding-left: max(env(safe-area-inset-left, 0), 1.5rem);
+  padding-right: max(env(safe-area-inset-right, 0), 1.5rem);
+
+  &.MuiListItem-root.Mui-selected,
+  &.MuiListItem-root.Mui-selected:hover {
+    background-color: revert;
+  }
+`;
+const StyledListItemText = styled(ListItemText).attrs({ disableTypography: true })`
+  display: contents;
+`;
+
+interface SelectOption {
+  label: ReactNode;
+  value: string;
+}
+
+type SelectItemProps = SelectOption & ListItemProps;
+const SelectItem = ({ label, selected, value, ...listItemProps }: SelectItemProps) => (
+  <StyledListItem {...listItemProps}>
+    <StyledListItemText primary={label} />
+    {selected && <CheckIcon htmlColor={useTheme().palette.primary.main} width={24} height={24} />}
+  </StyledListItem>
+);
+
+type FullScreenSelectProps = Pick<
+  SelectProps,
+  'children' | 'className' | 'id' | 'label' | 'onClose' | 'onOpen' | 'open'
+> & {
+  defaultValue?: string | number | null;
+  onChange?: (
+    event: ChangeEvent<{ name?: string | undefined; value: string | number | null }>,
+  ) => void;
+  value?: string | number | null;
+};
+
+export const FullScreenSelect = ({
+  children,
+  defaultValue,
+  label = 'Select an option',
+  onChange,
+  onClose,
+  value,
+}: FullScreenSelectProps) => {
+  const [selectedItem, setSelectedItem] = useState<typeof value>(
+    value !== null ? value : defaultValue,
+  );
+  useEffect(() => setSelectedItem(value), [value]);
```

Comment:
As far as I can tell, this local state and useEffect for selectedItem is redundant, as we can just use value directly below to find the selected item.